src/db/db.ts

```diff
@@ -0,0 +1,81 @@
+import { Observable, of } from 'rxjs';
+import { Sequelize, ModelCtor } from 'sequelize';
+import { CexOrder, CexOrderInstance } from './cex-order';
+import { CexFee, CexFeeInstance } from './cex-fee';
+import { CexTrade, CexTradeInstance } from './cex-trade';
+import { Logger } from '../logger';
+import { mergeMap } from 'rxjs/operators';
+
+type InitDBparams = {
+  dataDir?: string;
+  logger: Logger;
+};
+
+type InitDBResponse = {
+  CexOrder: ModelCtor<CexOrderInstance>;
+  CexFee: ModelCtor<CexFeeInstance>;
+  CexTrade: ModelCtor<CexTradeInstance>;
+};
+
+const createModels = (sequelize: Sequelize): InitDBResponse => {
+  const models = {
+    CexOrder: CexOrder(sequelize),
+    CexFee: CexFee(sequelize),
+    CexTrade: CexTrade(sequelize),
+  };
+
+  models.CexOrder.hasMany(models.CexTrade, {
+    as: 'cexOrderTrades',
+    foreignKey: 'orderId',
+    constraints: true,
+  });
+  models.CexTrade.belongsTo(models.CexOrder, {
+    as: 'cexOrderInstance',
+    constraints: true,
+    foreignKey: 'orderId',
+  });
+
+  models.CexFee.belongsTo(models.CexOrder, {
+    foreignKey: 'orderId',
+    constraints: true,
+  });
+  models.CexFee.belongsTo(models.CexTrade, {
+    foreignKey: 'tradeId',
+    constraints: true,
+  });
+
+  return models;
+};
+
+const initDB$ = ({
+  logger,
+  dataDir,
+}: InitDBparams): Observable<InitDBResponse> => {
+  const sequelize = new Sequelize({
+    storage: dataDir ? `${dataDir}/arby.db` : undefined,
+    logging: logger.trace,
+    dialect: 'sqlite',
+  });
+
+  return of(createModels(sequelize)).pipe(
```

Comment:
What's going on here? Did you mean to use `from` operator here instead?

Reply:
Uhm, there's actually a sequence here that has to be executed sequentially, so first authenticate then model's sync promises in the given order. I changed `of` to `from` for promises but still can't see a way to shorten it.

Reply:
Any advice would be appreciated